File: codeforces/problemset/1138A/main.py
_ = int(input())
t = [int(x) for x in input().split()]
previous = t.pop(0)
ans = 0
counter = [0, 0, 0]
counter[previous] = 1
for i in t:
    if previous != i:
        ans = max(ans, min(counter[1], counter[2]) * 2)
        counter[i] = 0
    counter[i] += 1
    previous = i
ans = max(ans, min(counter[1], counter[2]) * 2)
print(ans)

File: codeforces/problemset/1138A/main.cpp
#include<iostream>

using namespace std;

int main(int argc, char const *argv[])
{
    int n;
    cin >> n;
    int t[n];
    for (int i=0; i<n; i++) {
        cin >> t[i];
    }
    int previous = t[0];
    int ell = 0;
    int tuna = 0;
    if (previous == 1) tuna++;
    if (previous == 2) ell++;
    int flag = 0;
    int ans = 0;
    // 1 -> tuna; 2 -> ell
    for (int i=1; i<n; i++) {
        if (previous != t[i]) {
            ans = max(ans, min(ell, tuna) * 2);
            if (t[i] == 1) tuna = 0; 
            if (t[i] == 2) ell = 0;
        }
        if (t[i] == 1) tuna++;
        if (t[i] == 2) ell++;
        previous = t[i];
    }
    ans = max(ans, min(ell, tuna) * 2);
    cout << ans << endl;
    return 0;
}
